### ISSUE TITLE
com.intellij.diagnostic.PluginException: 314 ms to call on EDT FetchAllRemotesAction#update@GoToAction (com.virtuslab.gitmachete.frontend.actions.toolbar.FetchAllRemotesAction). Revise AnAction.getActionUpdateThread property [Plugin: com.virtuslab.git-machete]

### ISSUE THREAD

**Issue by ekitrak** [VirtusLab/git-machete-intellij-plugin#1457]
## Running environment
- Git Machete plugin version - 3.3.1
- IDE - IntelliJ IDEA 2022.3.1
- Operating system - Windows 10 10.0

## Bug description
**Please include steps to reproduce (like `go to...`/`click on...` etc.) + expected and actual behaviour**.

## (optional) Screenshots
If applicable, add screenshots (or screen recordings, see [Peek](https://github.com/phw/peek#peek---an-animated-gif-recorder) on Linux)
to help explain your problem.

## IDEA - additional info
N/A

## IDEA - stack trace
```
com.intellij.diagnostic.PluginException: 314 ms to call on EDT FetchAllRemotesAction#update@GoToAction (com.virtuslab.gitmachete.frontend.actions.toolbar.FetchAllRemotesAction). Revise AnAction.getActionUpdateThread property [Plugin: com.virtuslab.git-machete]
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:23)...
	Suppressed: java.lang.Throwable: EDT-trace-at-209-ms
		at java.base@17.0.5/java.lang.ClassLoader.findLoadedClass0(Native Method)...
	Suppressed: java.lang.Throwable: EDT-trace-at-162-ms
		at java.base@17.0.5/jdk.internal.misc.Unsafe.park(Native Method)
		at java.base@17.0.5/java.util.concurrent.locks.LockSupport.park(LockSupport.java:341)
		at java.base@17.0.5/java.util.concurrent.ForkJoinTask.awaitDone(ForkJoinTask.java:468)
		at java.base@17.0.5/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:687)
		at java.base@17.0.5/java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159)
		at java.base@17.0.5/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:173)
		at java.base@17.0.5/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
		at java.base@17.0.5/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
		at java.base@17.0.5/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:765)
		at org.reflections.Reflections.scan(Reflections.java:173)
		at org.reflections.Reflections.<init>(Reflections.java:126)
		at org.reflections.Reflections.<init>(Reflections.java:159)
		at org.reflections.Reflections.<init>(Reflections.java:151)
		at com.virtuslab.binding.RuntimeBinding.<clinit>(RuntimeBinding.java:14)
		at com.virtuslab.gitmachete.frontend.ui.services.SelectedGitRepositoryService.<init>(SelectedGitRepositoryService.java:23)
		at java.base@17.0.5/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
		at java.base@17.0.5/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
		at java.base@17.0.5/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
		at java.base@17.0.5/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
		at java.base@17.0.5/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
		at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:970)
		at com.intellij.serviceContainer.ComponentManagerImpl.createLightService(ComponentManagerImpl.kt:929)
		at com.intellij.serviceContainer.ComponentManagerImpl.getOrCreateLightService$lambda$9$lambda$7(ComponentManagerImpl.kt:737)
		at com.intellij.serviceContainer.ComponentManagerImpl$$Lambda$1450/0x0000000101130c70.run(Unknown Source)
		at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeNonCancelableSection$3(CoreProgressManager.java:224)
		at com.intellij.openapi.progress.impl.CoreProgressManager$$Lambda$554/0x0000000100679ee0.compute(Unknown Source)
		at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:664)
		at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:620)
		at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$computeInNonCancelableSection$4(CoreProgressManager.java:232)
		at com.intellij.openapi.progress.impl.CoreProgressManager$$Lambda$556/0x000000010067b528.compute(Unknown Source)
		at com.intellij.openapi.progress.Cancellation.computeInNonCancelableSection(Cancellation.java:99)
		at com.intellij.openapi.progress.impl.CoreProgressManager.computeInNonCancelableSection(CoreProgressManager.java:232)
		at com.intellij.openapi.progress.impl.CoreProgressManager.executeNonCancelableSection(CoreProgressManager.java:223)
		at com.intellij.serviceContainer.ComponentManagerImpl.getOrCreateLightService(ComponentManagerImpl.kt:736)
		at com.intellij.serviceContainer.ComponentManagerImpl.doGetService(ComponentManagerImpl.kt:685)
		at com.intellij.serviceContainer.ComponentManagerImpl.getService(ComponentManagerImpl.kt:629)
		at com.virtuslab.gitmachete.frontend.actions.base.BaseProjectDependentAction.getSelectedGitRepository(BaseProjectDependentAction.java:68)
		at com.virtuslab.gitmachete.frontend.actions.toolbar.FetchAllRemotesAction.onUpdate(FetchAllRemotesAction.java:43)
		at com.virtuslab.gitmachete.frontend.actions.base.BaseProjectDependentAction.update(BaseProjectDependentAction.java:40)
		at com.intellij.openapi.actionSystem.ex.ActionUtil.lambda$performDumbAwareUpdate$0(ActionUtil.java:153)
		at com.intellij.openapi.actionSystem.ex.ActionUtil$$Lambda$1346/0x0000000101031708.run(Unknown Source)
		at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareUpdate(ActionUtil.java:176)
		at com.intellij.openapi.actionSystem.impl.ActionUpdater.doUpdate(ActionUpdater.java:720)
		at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$updateActionReal$4(ActionUpdater.java:136)
		at com.intellij.openapi.actionSystem.impl.ActionUpdater$$Lambda$1953/0x00000001015dbd90.get(Unknown Source)
		at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$computeOnEdt$7(ActionUpdater.java:233)
		at com.intellij.openapi.actionSystem.impl.ActionUpdater$$Lambda$2193/0x000000010172a9e0.compute(Unknown Source)
		at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:68)
		at com.intellij.openapi.progress.ProgressManager$$Lambda$1877/0x0000000101588000.run(Unknown Source)
		at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:188)
		at com.intellij.openapi.progress.impl.CoreProgressManager$$Lambda$1863/0x0000000101580000.run(Unknown Source)
		at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:589)
		at com.intellij.openapi.progress.impl.CoreProgressManager$$Lambda$1865/0x00000001015806e8.compute(Unknown Source)

**Comment by PawelLipski** [comment]
Whoops good to know... lemme see what's this heavy lifting that we put onto UI thread...

**Comment by PawelLipski** [comment]
Okay, #1458 should provide a fix.

**Comment by PawelLipski** [comment]
To be released soon in v3.3.2
